src/middleware.ts

```diff
@@ -57,6 +57,19 @@ export default authMiddleware({
 
         return NextResponse.redirect(`${req.nextUrl.href}criar-empresa`);
       }
+      if (
+        req.nextUrl.pathname === `/criar-empresa` &&
+        req.cookies.get(`__session`)
+      ) {
+        const user = await clerkClient.users.getUser(`${auth.userId}`);
+
+        if (user.publicMetadata.slug) {
+          req.nextUrl.pathname = `/`;
```

Comment:
This line is necessary?

Reply:
Yes, because if you type "/criar-empresa" and press enter, your `req.nextUrl.href = "/criar-empresa"` . So when redirecting the page, the redirect will be `/criar-empresaconfiguracoes/user/id`.